ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/EventsRange.scala

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.daml.platform.store.dao.events
+
+import anorm.SqlParser.get
+import anorm.{RowParser, SqlStringInterpolation}
+import com.daml.ledger.participant.state.v1.Offset
+import scalaz.syntax.std.option._
+
+final case class EventsRange[A](startExclusive: A, endInclusive: A)
+
+object EventsRange {
+  private val EmptyRowIdRange = EventsRange(0L, 0L)
+
+  private implicit val `offset to statement converter` =
+    com.daml.platform.store.Conversions.OffsetToStatement
+
+  private val rangeParser: RowParser[EventsRange[Long]] =
+    (get[Option[Long]]("start") ~ get[Option[Long]]("end")).map { row =>
+      EventsRange(
+        startExclusive = row._1.getOrElse(EmptyRowIdRange.startExclusive),
+        endInclusive = row._2.getOrElse(EmptyRowIdRange.endInclusive))
+    }
+
+  private val endParser: RowParser[EventsRange[Long]] =
+    get[Option[Long]]("end").map { end =>
+      end.cata(
+        x => EmptyRowIdRange.copy(endInclusive = x),
+        EmptyRowIdRange
+      )
+    }
```

Comment:
```suggestion
    get[Option[Long]]("end").map(
      _.fold(EmptyRowIdRange)(end => EmptyRowIdRange.copy(endExclusive = end)
    )
```
You can very easily replace this with a native `fold` or `map`/`getOrElse` as you have done in `rangeParser`. I'd prefer not bringing Scalaz into the picture for a single use that can easily be covered by the standard library.